docs/source/array-creation.rst

```diff
@@ -201,6 +201,42 @@ on your Dask array to turn in to a normal NumPy array:
    array([0, 1, 4, 9, 16, 25])
 
 
+Intermediate storage
+--------------------
+
+.. autosummary::
+   store
+
+In some cases, one may wish to store an intermediate value that will be used in
+later computations. This could be useful for a variety of cases like saving
+work before a flaky step, as a simple form of caching, a useful
+restore/stopping point in a long computation, inclusion of the intermediate
+value in later analysis, etc.
```

Comment:
I think that many people reading this paragraph might also benefit from looking at `persist`.  I'm concerned that this text might direct people-that-should-be-using-persist to `store` instead.

Reply:
Hmm...not sure I follow. The use cases listed above include cases where having the data permanently stored is a requirement. Are there particular cases that lead you to this conclusion? Could you propose some alternative wording to better capture these cases?

Reply:
I think that we should probably add a small section to the In Memory section just above that includes `persist` (my guess is that we wrote those docs before persist).  I think that we should expand the title of the Intermediate Storage section to say "Intermediate Storage on Disk" or something similar.

Similarly this sentence

> In some cases, one may wish to store an intermediate value that will be used in later computations

Might be expanded to include the following

```diff
-In some cases, one may wish to store an intermediate value that will be used in later computations
+In some cases, one may wish to store an intermediate value to disk that will be used in later computations
```

Reply:
Adding a section on `persist` above seems like a good idea. Having a page with `compute`, `persist`, and `store` outlined is generally useful IMHO even if it is somewhat orthogonal to this PR.

I'm a little wary of saying "on disk" as that is not strictly a requirement of a store. For example some tests show how this can be done with a NumPy array and some users leverage this ( https://github.com/dask/dask/issues/3595 ). Independently things like Zarr allow storage in Redis ( https://github.com/zarr-developers/zarr/pull/372 ).

Reply:
Maybe we can reuse `store`'s existing terminology and say...

> ...store intermediate values to `targets`...

?

Reply:
Happy to avoid the term "disk".  Maybe "long term storage"?  At this point in their education I don't expect readers to be familar with the terminology of the `store` function.